File: PlacementCellBackend/Services/Interfaces/IRestaurantService.cs
using PlacementCellBackend.Models;

namespace PlacementCellBackend.Services.Interfaces
{
    public interface IRestaurantService
    {
        Task<IEnumerable<Restaurents>> GetAllRestaurantsAsync();
        Task<Restaurents?> GetRestaurantByIdAsync(int id);
        Task<Restaurents> CreateRestaurantAsync(Restaurents restaurant);
        Task<bool> UpdateRestaurantAsync(int id, Restaurents restaurant);
        Task<bool> DeleteRestaurantAsync(int id);
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/services/Interfaces/IRestaurantService.java
package com.example.placementicsbackend.services.Interfaces;

import com.example.placementicsbackend.models.Restaurents;
import java.util.List;

public interface IRestaurantService {

    List<Restaurents> getAllRestaurants();

    Restaurents getRestaurantById(int id);

    Restaurents createRestaurant(Restaurents restaurant);

    boolean updateRestaurant(int id, Restaurents restaurant);

    boolean deleteRestaurant(int id);
}
